### PR TITLE
Spotbugs plugin update to current version 4.7.1

### PR DESCRIPTION
**Pull request by EdColeman**
Updating spotbugs to current release. This does not change the error check level

Fixes new error 
```
GarbageCollectionLogger.java:[line 117] SSD_DO_NOT_USE_INSTANCE_LOCK_ON_SHARED_STATIC_DATA
```

### CODE REVIEW

**Review by Manno15 — APPROVED** [review]

**Review by milleruntime — CHANGES_REQUESTED** [review]
I think you can just make `lastMemoryCheckTime` a private member of the class.

**Comment by EdColeman** [comment]
Making `lastMemoryCheckTime` private changes the current behavior.  Not sure if that is necessary - but seemed the best course of action was to preserve the current behavior and then consider refactoring.  If there are multiple instances created in a server, the current behavior is that the log / check will run once per defined period even if there were multiple instance created.

It may be desirable to lift the GCLogger to AbstractSever so that it is initialized / runs once per server process without static and possible synchronization required - but that seemed outside of the scope of this PR.

**Review by ctubbsii — APPROVED** [review]
I think it's fine to preserve current behavior and fix the spotbugs issue.

File: pom.xml [hunk 1]
```diff
@@ -720,7 +720,7 @@
         <plugin>
           <groupId>com.github.spotbugs</groupId>
           <artifactId>spotbugs-maven-plugin</artifactId>
-          <version>4.5.3.0</version>
+          <version>4.7.1.0</version>
```
Comment (ctubbsii) [inline pom.xml:723]: I had been working on getting us to 4.7.0.0. It generated a bunch of new warnings. If this passes, I'm wondering if they rolled some of those changes back a bit for 4.7.1.0, due to them being too aggressive?

**Review by ctubbsii — APPROVED** [review]

**Pull request merged**